src/core/src/op/bitwise_not.cpp

```diff
@@ -0,0 +1,33 @@
+// Copyright (C) 2018-2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+#include "openvino/op/bitwise_not.hpp"
+
+#include "itt.hpp"
+#include "openvino/core/validation_util.hpp"
+#include "openvino/op/op.hpp"
+
+namespace ov {
+namespace op {
+namespace v13 {
+BitwiseNot::BitwiseNot(const Output<Node>& arg) : op::Op({arg}) {
+    constructor_validate_and_infer_types();
+}
+void BitwiseNot::validate_and_infer_types() {
+    OV_OP_SCOPE(v13_BitwiseNot_validate_and_infer_types);
+    const auto& element_type = get_input_element_type(0);
+    NODE_VALIDATION_CHECK(this,
+                          element_type.is_dynamic() || element_type.is_integral() || element_type == element::boolean,
+                          "The element type of the input tensor must be integral or boolean.");
```

Comment:
In fact `bool` type is also considered as integral type 
(it complies with cpp `std::is_integral` https://en.cppreference.com/w/cpp/types/is_integral)
This is why there is a separate check for `is_integral_number`:
https://github.com/openvinotoolkit/openvino/blob/b6427ad0fb731b82d352df6db7d9712278179ffb/src/core/include/openvino/core/type/element_type.hpp#L76-L79 https://github.com/openvinotoolkit/openvino/blob/b6427ad0fb731b82d352df6db7d9712278179ffb/src/core/src/type/element_type.cpp#L366-L367
So the additional check for `|| element_type == element::boolean` is currently redundant:
```suggestion
    NODE_VALIDATION_CHECK(this,
                          element_type.is_dynamic() || element_type.is_integral(),
                          "The element type of the input tensor must be integer or boolean.");
```

Reply:
Thanks, removed redundant check